src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReleaseRepository.cs

```diff
@@ -75,46 +71,35 @@ public class ReleaseRepository : IReleaseRepository
         public async Task<Guid> CreateStatisticsDbReleaseAndSubjectHierarchy(Guid releaseId)
         {
             var release = await _contentDbContext.Releases
-                .Include(r => r.Publication)
+                .FirstAsync(r => r.Id == releaseId);
+
+            var existingStatsRelease = await _statisticsDbContext.Release
                 .FirstOrDefaultAsync(r => r.Id == releaseId);
 
-            var existingRelease =
-                await _statisticsDbContext.Release.FindAsync(release.Id);
-            if (existingRelease == null)
+            if (existingStatsRelease == null)
             {
-                var statsRelease = _mapper.Map(release, new Data.Model.Release());
-                await _statisticsDbContext.Release.AddAsync(statsRelease);
+                _statisticsDbContext.Release.Add(new Data.Model.Release
+                {
+                    Id = releaseId,
+                    PublicationId = release.PublicationId
+                });
             }
             else
             {
-                _mapper.Map(release, existingRelease);
-                _statisticsDbContext.Release.Update(existingRelease);
+                existingStatsRelease.PublicationId = release.PublicationId;
+                _statisticsDbContext.Release.Update(existingStatsRelease);
```

Comment:
I know you didn't add this, but I don't understand why this `else` is necessary. I think it can be removed.

`CreateStatisticsDbReleaseAndSubjectHierarchy` looks like it is only called when uploading a new file or zip. If the Statistics DB Release entry already exists, we shouldn't need to update it, I think?

There is no way to change the publication for a release, so the publication shouldn't ever change.

Reply:
I was also confused about this. We shouldn't need to update the publication so I've removed it. Thanks.